telemetrydeck.go

```diff
@@ -280,23 +298,26 @@ func (c *Client) SendSignal(ctx context.Context, signalType string, payload map[
 	if err != nil {
 		return err
 	}
-
 	request.Header.Set("Content-Type", "application/json; charset=utf-8")
 
-	response, err := c.httpClient.Do(request)
-	if err != nil {
-		return err
-	}
-	defer response.Body.Close()
-
-	if c.testMode && response.StatusCode >= 400 {
-		log.Printf("response status: %d", response.StatusCode)
-		log.Printf("request body: %s", b)
-		bodyBytes, err := io.ReadAll(response.Body)
-		if err == nil {
-			log.Printf("details: %s", string(bodyBytes))
+	go func(r *http.Request) {
+		response, err := c.httpClient.Do(request)
+		if err != nil {
+			if c.logger != nil {
+				c.logger.Printf("error submitting HTTP request: %s", err)
+			}
 		}
-	}
+		defer response.Body.Close()
+
+		if response.StatusCode >= 400 && c.testMode && c.logger != nil {
```

Comment:
```suggestion
		if response.StatusCode >= 400 || c.testMode && c.logger != nil {
```

Did you really mean AND here?

Reply:
Yes I did. c.logger MUST be defined. And I only wanted to log these if testMode is active (not in production).